src/functions/template-tags/html.php

```diff
@@ -132,3 +132,83 @@ function tribe_disabled( $disabled, $echo = true ) {
 		}
 	}
 }
+
+function tribe_format_field_dependency( $deps ) {
+    // Sanity check.
+    if ( empty( $deps ) ) {
+        return '';
+	}
+
+	// Let's be case-insensitive!
+	if ( isset( $deps['ID'] ) ) {
+		$deps['id'] = $deps['ID'];
+		unset( $deps['ID'] );
+	}
+
+    // No ID to hook to? Bail.
+    if ( empty( $deps['id'] ) ) {
+        return;
+    }
+
+    $dependency = '';
+
+    $accepted = [
+        'id',
+        'is',
+        'is-not',
+        'is-empty',
+        'is-not-empty',
+        'is-numeric',
+        'is-not-numeric',
+        'is-checked',
+        'is-not-checked',
+    ];
+
+    foreach ( $deps as $attr => $value ) {
+        // Attributes are always lower case.
+        $attr = strtolower( $attr );
+
+        // Valid dependency check.
+        // Note this means the dependency has to begin with "is", except for "id"!
+        if ( ! in_array( $attr, $accepted ) ) {
```

Comment:
This is a good example of "filtering while interpolating" that could be avoided.

```php
$valid_deps = array_intersect_key( $deps, array_flip( $accepted ) );
```

And then iterate on the result; it's a small optimization, but it makes for leaner code and does not move a full array around.